### PR TITLE
Login redux

### PR DESCRIPTION
**Pull request by CristianRWolox**
## Summary
- Add action for authentication
- Add reducer for authentication
- Implement service to authenticate 

## Trello Card
https://trello.com/c/JlFoD0oE/33-n-103-funcionalidad-de-login-con-redux

### CODE REVIEW

File: wbooks/src/redux/auth/actions.ts [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+import { createTypes, completeTypes } from 'redux-recompose';
+
+const completedActions = completeTypes({ primaryActions: ['LOGIN'] });
+
+export const actions = createTypes(completedActions, '@@AUTH');
+
+const actionCreators = {
+  login: (email: string, password: string) => ({
+    type: actions.LOGIN,
+    target: 'user',
```
Comment (guidoprinc) [inline wbooks/src/redux/auth/actions.ts]: extract target as a constant outside `actionCreators`

File: wbooks/src/services/AuthService.ts [hunk 2]
```diff
@@ -0,0 +1,7 @@
+interface Props {
+  email: string;
+  password?: string;
+}
+
+export const login = ({ email }: Props) =>
+  Promise.resolve({ ok: true, data: { id: 1, toke: 'AE12OI45PK56LK', client: email }, problem: '' });
```
Comment (guidoprinc) [inline wbooks/src/services/AuthService.ts]: token

**Review by guidoprinc — CHANGES_REQUESTED** [review]

File: wbooks/src/app/screens/Login/index.tsx [hunk 3]
```diff
@@ -1,17 +1,28 @@
...
+  password: string;
+}
+
 const Login = () => {
   const { control, handleSubmit, errors } = useForm();
   const navigation = useNavigation();
+  const dispatch = useDispatch();
 
-  const onSubmit = () => navigation.navigate('Library');
+  const onSubmit = (data: DataForm) => {
```
Comment (mcolombo87) [inline wbooks/src/app/screens/Login/index.tsx]: extract email and password from data.

```suggestion
  const onSubmit = (data: DataForm) => {
```
```suggestion
  const onSubmit = ({ email, password }: DataForm) => {
```

File: wbooks/src/services/AuthService.ts (hunk 2 above)
Comment (CristianRWolox) [reply wbooks/src/services/AuthService.ts]: Done!

File: wbooks/src/redux/auth/actions.ts (hunk 1 above)
Comment (CristianRWolox) [reply wbooks/src/redux/auth/actions.ts]: Done!

File: wbooks/src/app/screens/Login/index.tsx (hunk 3 above)
Comment (CristianRWolox) [reply wbooks/src/app/screens/Login/index.tsx]: Done!

**Review by guidoprinc — APPROVED** [review]

**Pull request merged**